app/src/main/java/com/github/se/eduverse/ui/quizz/QuizzScreen.kt

```diff
@@ -110,14 +112,25 @@ fun QuizScreen(navigationActions: NavigationActions, quizzRepository: QuizzRepos
                   onClick = {
                     focusManager.clearFocus()
                     scope.launch {
+                      isLoading = true
+                      isQuizSubmitted = false
+                      selectedAnswers.clear()
+                      errorMessage = null
                       try {
-                        isQuizSubmitted = false
-                        questions =
+                        val fetchedQuestions =
                             quizzRepository.getQuestionsFromGPT(
                                 topic, difficulty, numberOfQuestions)
-                        selectedAnswers.clear()
+                        if (fetchedQuestions.isNotEmpty()) {
+                          questions = fetchedQuestions
+                        } else {
+                          errorMessage = "Ouchhh! the AI failed to generate the quiz. Try again."
+                        }
```

Comment:
adding error messages is good, well done! but it could be better to add more detailed error messages for different types of errors (e.g. connection failure or internal error).

Reply:
I think that it's not a idea for user experience, users are not supposed to know in detail the problem that occurred.

Reply:
for user experience, I think It s better to know if the problem is for example an API call error (it is not the user's fault), or if it is an error that is the user's fault.

Reply:
Thank you for your answer, I understand your point. However when designing apps the functionality needs to be the most minimalist as possible from user point of view. Indeed a student of 12 years using this app as a tool may not know technical concepts as API's etc..., that's why I think we should stay simple to enhance experience for users. Moreover it could cause some security issues.